### PR TITLE
Feature: Knowledge Base APIs

### PR DESCRIPTION
**Pull request by subintp**
# Pull Request Template

## Description

```
I saw a sign at a gas station. It said 'help wanted'. There was another sign below it that said 'self-service'. 
So I hired myself. Then I made myself the boss. I gave myself a raise. I paid myself. Then I quit.
-  Steven Wright 
```

Addresses: #714 

Introduces models and APIs for
  - kbase portals
  - kbase categories
  - kbase folders 
  - kbase articles

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by RonWalker22** [comment]
@subintp Do you plan on displaying the Knowledge Base within the website widget? 

If so, how are you going to do that?

I'm curious because I'm building a different type of portal. 

Previously I just created a new route, because the widget was already using vue-router. But vue-router has been replaced recently in favor of this file [Router.vue](https://github.com/chatwoot/chatwoot/blob/develop/app/javascript/widget/views/Router.vue). 

I'm going to do a workaround and just bring back vue-router in my fork. But wanted to know if you have a better idea.

**Comment by tabrza** [comment]
hey, i was curious if this was going into the next merge on develop or if there's a decision to pause on this?

**Comment by sojan-official** [comment]
@tabrza there is no decision to pause this, the team got busy with some other priorities.  This feature going live in with v1.9 might be tricky since there is a lot of work left. But we could keep the expectation up for 1.10/1.11 . 

@RonWalker22 yes we have plans to integrate the portal into the chatwidget.  

@pranavrajs / @nithindavid  could you talk on the vue-router ?

File: app/models/kbase/article.rb [hunk 1]
```diff
@@ -24,11 +23,10 @@ class Kbase::Article < ApplicationRecord
 
   validates :account_id, presence: true
   validates :category_id, presence: true
-  validates :folder_id, presence: true
   validates :author_id, presence: true
 
   validates :title, presence: true
   validates :content, presence: true
 
-  enum status: { unpublished: 0, published: 1 }
```
Comment (pranavrajs) [inline app/models/kbase/article.rb]: should we add archived as well?

**Review by sojan-official — APPROVED** [review]

**Pull request merged**